src/ck-ldb/LBManager.C

```diff
@@ -922,66 +920,71 @@ int LBRegisterObjUserData(int size)
   return CkpvAccess(lbobjdatalayout).claim(size);
 }
 
-
-class client {
-  friend class LocalBarrier;
-  void* data;
-  LDResumeFn fn;
+class LBClient {
+public:
+  Chare* chare;
+  std::function<void()> fn;
   int refcount;
+
+  LBClient(Chare* chare, std::function<void()> fn, int refcount) :
+    chare(chare),
+    fn(fn),
+    refcount(refcount)
+  {}
 };
-class receiver {
-  friend class LocalBarrier;
-  void* data;
-  LDBarrierFn fn;
+
+class LBReceiver {
+public:
+  std::function<void()> fn;
   int on;
+
+  LBReceiver(std::function<void()> fn, int on = 1) :
+    fn(fn),
+    on(on)
+  {}
 };
 
-void LBManager::AddClients(Chare* _new_chare, bool atsync, bool atsync_notify) {
-  //Chare types: AtSync, reached-notify, resume-notify
-  _new_chare->r_count = localBarrier.cur_refcount;
-  _new_chare->atsync = atsync;
-  _new_chare->atsync_notify = atsync_notify;
-  if(_new_chare->atsync)
-  localBarrier.client_count++;
+LDBarrierClient LocalBarrier::AddClient(Chare* chare, std::function<void()> fn) {
+  LBClient* new_client = new LBClient(chare, fn, cur_refcount);
```

Comment:
Any reason why we have LBClient and LDBarrierClient? Seems confusing for little benefit.

Reply:
As I said below, `LBClient` and `LBReceiver` are only intended to be used internally.

I do think that this whole shebang should be renamed and restructured to something more sensible, but I think I'm going to punt on that for now and defer it for the change that extracts `LocalBarrier` from being so intertwined with the load balancing infrastructure.

Reply:
Created #2779 for this.